manifests/istio-control/istio-discovery/templates/autoscale.yaml

```diff
@@ -1,8 +1,9 @@
+{{ if or (eq .Values.revision "") (not .Values.clusterResources) }}
 {{- if and .Values.pilot.autoscaleEnabled .Values.pilot.autoscaleMin .Values.pilot.autoscaleMax }}
 apiVersion: autoscaling/v2beta1
 kind: HorizontalPodAutoscaler
 metadata:
-  name: istio-pilot{{ .Values.version }}
+  name: istio-pilot{{ .Values.revision }}
```

Comment:
It makes the charts a bit more verbose, but can we change this to `{{if .Values.revision}}-{{.Values.revision }}{{end}}` everywhere? Then this will render as `istio-pilot-stable`, which I think is a bit cleaner

Reply:
Can we do it in a separate PR that just introduces the - ?

Not sure what's the cleanest way - maybe a macro to create a 'suffix'